### DISCUSSION TITLE
Full Range vs Narrow Range Key

### DISCUSSION THREAD

**Discussion by ryanpreller** [aws/aws-cdi-sdk#77] [General] [answered]
It is not uncommon to allow configuration of key as full range on SDI equipment (eg. either to scale narrow to full, scale full to narrow, or to clip full to narrow), while video remains narrow range.

Is it implied by the specification through CdiAvmVideoRange that key should always be narrow range if video is also narrow range?

Is it also implied by the specification that video is always shaped by key, or is this also up to the individual vendor?

**Comment by mhhen** [comment, marked as answer]
Since the AVM specification does not contain a separate value for key, it is implied that value should match the one specified for the video range. Possible values are in [CdiAvmVideoRange](https://github.com/aws/aws-cdi-sdk/blob/eba76e328adebf2bfdf2ebdc9d0ef76e074eda29/include/cdi_baseline_profile_01_00_api.h#L92).

The latest AVM specification is at: https://cdi.elemental.com/specs/baseline-video-vxx.yy.html
